### PR TITLE
Fix invisible player when the attached entity is removed

### LINKED ISSUE #1467: Player invisible after removing of attached entity
Before:
http://i.imgur.com/gmhFBbD.png
After:
http://i.imgur.com/hsRfFFP.png

This happens when

player:set_detach()

is not called before removing the entity.

### PR DESCRIPTION
**Pull request by TeTpaAka**
Makes the parent aware of its children and detaches them on delete.
Fixes  #1467

### CODE REVIEW

File: src/script/lua_api/l_object.cpp [hunk 1]
```diff
@@ -132,10 +132,22 @@ int ObjectRef::gc_object(lua_State *L) {
 int ObjectRef::l_remove(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
+	GET_ENV_PTR;
+
 	ObjectRef *ref = checkobject(L, 1);
 	ServerActiveObject *co = getobject(ref);
-	if(co == NULL) return 0;
-	if(co->getType() == ACTIVEOBJECT_TYPE_PLAYER) return 0;
+	if(co == NULL)
```
Comment (est31) [inline src/script/lua_api/l_object.cpp]: space

File: src/script/lua_api/l_object.cpp [hunk 2]
```diff
@@ -132,10 +132,22 @@ int ObjectRef::gc_object(lua_State *L) {
...
+		return 0;
+	if(co->getType() == ACTIVEOBJECT_TYPE_PLAYER)
```
Comment (est31) [inline src/script/lua_api/l_object.cpp]: space

**Comment by est31** [comment]
I dunno, perhaps we can merge children management with GenericCAO, where its already done?

File: src/script/lua_api/l_object.cpp (hunk 2 above)
Comment (TeTpaAka) [inline src/script/lua_api/l_object.cpp]: fixed

File: src/script/lua_api/l_object.cpp (hunk 1 above)
Comment (TeTpaAka) [inline src/script/lua_api/l_object.cpp]: Fixed.

**Comment by TeTpaAka** [comment]
I don't know. I can make it, if you want, but I'm not sure if this is right. It seems like there were many common fields that aren't shared.

File: src/script/lua_api/l_object.cpp (hunk 1 above)
Comment (Zeno-) [inline src/script/lua_api/l_object.cpp]: Space, the new frontier. These are the voyages of the starship Enterprise. Its five-year missio...

Err... sorry. Ignore me, I was confused with something else.

**Comment by est31** [comment]
8d03301138c0e82728d61b957ea47a5a4061633a

**Pull request closed without merging**